### PR TITLE
Create user app

### PR DESCRIPTION
**Pull request by shayakrach**
Creating a user model, a static method for creating users.
Adding tests for user creation.

### CODE REVIEW

**Comment by shayakrach** [comment]
HI @ormergi,
As I mentioned in #21, PyTest fails when jobborad is still there. 
That is why it should be in the same PR.

File: users/tests.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pytest
+from django.contrib.auth.models import User
+
+
+@pytest.mark.django_db
+def test_user_create():
+    User.objects.create_user('test', 'test@test.com', 'test')
+    count = User.objects.all().count()
+    print(count)
```
Comment (EdDev) [inline users/tests.py]: Printing is for debugging, but it should not remain in the code.

File: users/tests.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    assert User.objects.count() == 1
```
Comment (EdDev) [inline users/tests.py]: Before `User.objects.all().count()` was fetched but the final assert uses `User.objects.count()`.

File: users/tests.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+@pytest.mark.django_db
+def test_user_create1():
+    count = User.objects.all().count()
+    print(count)
```
Comment (EdDev) [inline users/tests.py]: same, prints should not remain in the published code.

File: users/tests.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+@pytest.mark.django_db
+def test_user_create():
+    User.objects.create_user('test', 'test@test.com', 'test')
+    count = User.objects.all().count()
+    print(count)
+    assert User.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_user_create1():
```
Comment (EdDev) [inline users/tests.py]: please name it per what is does. We are suppose to read the name and understand what it checks.

**Review by EdDev — DISMISSED** [review]
This PR is not rebased on master and the tests fail.

File: users/tests.py (hunk 1 above)
Comment (shayakrach) [reply users/tests.py]: Fixed.

File: users/tests.py (hunk 2 above)
Comment (shayakrach) [reply users/tests.py]: Fixed.

File: users/tests.py (hunk 3 above)
Comment (shayakrach) [reply users/tests.py]: Fixed.

File: users/tests.py (hunk 4 above)
Comment (shayakrach) [reply users/tests.py]: @EdDev We removed this test.
We will add more meaningful tests in another PR.

File: users/tests.py (hunk 1 above)
Comment (yuvalk) [reply users/tests.py]: and best is to use a logger
I recommend reading the documentation section about logging
https://docs.djangoproject.com/en/3.2/topics/logging/

Comment (shayakrach) [reply users/tests.py]: @yuvalk 
So should we use a logger instead?

File: users/models.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    role = models.CharField(
+        max_length=10,
+        choices=Role.choices,
+        default=Role.STUDENT,
+        blank=True,
+        null=True
+    )
+
+    @staticmethod
+    def create_user(username, email, password, first_name, last_name, role, team):
```
Comment (yuvalk) [inline users/models.py]: how can you set gender (for example)?

File: users/tests.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import pytest
+from django.contrib.auth.models import User
+
+
+@pytest.mark.django_db
+def test_user_create():
```
Comment (yuvalk) [inline users/tests.py]: you need to think about edge cases and add more tests
what happens if you try to add the same user twice?
what happens with "weird" usernames?
role that is longer than the designated 10 chars?
etc..

File: users/models.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (AshkenYariv) [inline users/models.py]: We need to remove team from here

File: users/models.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        django_user = DjangoUser.objects.create_user(username=username,
+                                                     email=email,
+                                                     password=password,
+                                                     first_name=first_name,
+                                                     last_name=last_name)
+        user = User(user=django_user,
+                    role=role,
+                    team=team)
```
Comment (AshkenYariv) [inline users/models.py]: also we need to remove team from here

File: users/tests.py (hunk 6 above)
Comment (AshkenYariv) [reply users/tests.py]: Alright tomorrow we will work on it

File: users/tests.py (hunk 1 above)
Comment (yuvalk) [reply users/tests.py]: I'd recommend using a logger, yes.
anyhow, dont leave debug prints in the code

**Comment by shayakrach** [comment]
Hi @yuvalk,

I added some tests.
Please tell me what else needs to be changed.

**Review by AshkenYariv — APPROVED** [review]
Looks good!

**Review by assafg95 — APPROVED** [review]

File: users/tests.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
+import pytest
+from users.models import Role, User
+from django.contrib.auth.models import User as DjangoUser
+
+
+@pytest.fixture
+def user_1():
```
Comment (yuvalk) [inline users/tests.py]: create_user_1 ?

**Comment by yuvalk** [comment]
I might still have some issues with the tests (might just be my misunderstandings)
approving
